### PR TITLE
[lldb] Make LanguageRuntime::GetTypeBitSize return an optional (NFC)

### PR DESCRIPTION
**Pull request by JDevlieghere**
Make LanguageRuntime::GetTypeBitSize return an optional. This should be NFC, though the ObjCLanguageRuntime implementation is (possibly) more defensive against returning 0.

I'm not sure if it's possible for both `m_ivar.size` and `m_ivar.offset` to be zero. Previously, we'd return 0 and cache it, only to discard it the next time when finding it in the cache, and recomputing it again. The new code will avoid putting it in the cache in the first place.

### CODE REVIEW

File: lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp [hunk 1]
```diff
@@ -4744,11 +4744,12 @@ TypeSystemClang::GetBitSize(lldb::opaque_compiler_type_t type,
...
-        if (objc_runtime) {
-          uint64_t bit_size = 0;
-          if (objc_runtime->GetTypeBitSize(GetType(qual_type), bit_size))
-            return bit_size;
+        if (ObjCLanguageRuntime *objc_runtime =
+                ObjCLanguageRuntime::Get(*process)) {
+          std::optional<uint64_t> bit_size =
+              objc_runtime->GetTypeBitSize(GetType(qual_type));
+          if (bit_size)
+            return *bit_size;
```
Comment (Michael137) [inline lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp:4752]: Couldn't this just be:
```suggestion
          return objc_runtime->GetTypeBitSize(GetType(qual_type));
```
?

**Review by Michael137 — APPROVED** [review]
LGTM! (minor nit)

File: lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp (hunk 1 above)
Comment (JDevlieghere) [reply lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp:4752]: That might be the right thing to do, but would be a change in behavior, as now, if it fails, we'll fall through to the `default` case first.

Comment (Michael137) [reply lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp:4752]: Aah you're right, missed the fallthrough here. Could fold the `GetTypeBitSize` call into the if-statement, like you did for `ObjCLanguageRuntime::Get`. But either way, LGTM

**Pull request merged**